utils/providers.py

```diff
@@ -267,6 +267,24 @@ def is_provider_setup(provider_key):
     return get_crud(provider_key).exists
 
 
+def existing_providers():
+    """Lists all providers that are already set up in the appliance."""
+    return filter(is_provider_setup, list_all_providers())
+
+
+def get_provider(provider_key):
+    """Returns a CRUD object of the provider key.
```

Comment:
This is a remainder of the older code. I think I will use the get_crud instead of this one.